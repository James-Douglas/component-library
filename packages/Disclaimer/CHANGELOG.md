# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.14.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.11...@comparethemarketau/manor-disclaimer@0.14.0) (2021-03-08)


### Features

* **various:** various fixes to resolve console warnings ([#602](https://github.com/comparethemarketau/manor-react/issues/602)) ([efbe8ff](https://github.com/comparethemarketau/manor-react/commit/efbe8ff4e4632d388f83bd6090b7ef116d450ebc))





## [0.13.11](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.10...@comparethemarketau/manor-disclaimer@0.13.11) (2021-03-05)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.10](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.9...@comparethemarketau/manor-disclaimer@0.13.10) (2021-02-26)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.8...@comparethemarketau/manor-disclaimer@0.13.9) (2021-02-26)


### Bug Fixes

* üêõ validation prop to accept nodes, arrays and strings ([#594](https://github.com/comparethemarketau/manor-react/issues/594)) ([657f23e](https://github.com/comparethemarketau/manor-react/commit/657f23e54af7f9374e1c7f89c60f7752bada1716))





## [0.13.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.7...@comparethemarketau/manor-disclaimer@0.13.8) (2021-02-26)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.6...@comparethemarketau/manor-disclaimer@0.13.7) (2021-02-26)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.5...@comparethemarketau/manor-disclaimer@0.13.6) (2021-02-24)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.4...@comparethemarketau/manor-disclaimer@0.13.5) (2021-02-23)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.3...@comparethemarketau/manor-disclaimer@0.13.4) (2021-02-17)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.2...@comparethemarketau/manor-disclaimer@0.13.3) (2021-02-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.1...@comparethemarketau/manor-disclaimer@0.13.2) (2021-02-11)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.13.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.13.0...@comparethemarketau/manor-disclaimer@0.13.1) (2021-02-09)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.13.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.6...@comparethemarketau/manor-disclaimer@0.13.0) (2020-12-18)


### Features

* **checkbox:** added boxed checkboxes ([#525](https://github.com/comparethemarketau/manor-react/issues/525)) ([0d61056](https://github.com/comparethemarketau/manor-react/commit/0d6105671e33ab8de139bafb62e8cdf41acfa735))





## [0.12.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.5...@comparethemarketau/manor-disclaimer@0.12.6) (2020-12-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.12.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.4...@comparethemarketau/manor-disclaimer@0.12.5) (2020-12-09)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.12.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.3...@comparethemarketau/manor-disclaimer@0.12.4) (2020-12-08)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.12.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.2...@comparethemarketau/manor-disclaimer@0.12.3) (2020-12-04)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.12.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.1...@comparethemarketau/manor-disclaimer@0.12.2) (2020-11-30)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.12.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.12.0...@comparethemarketau/manor-disclaimer@0.12.1) (2020-11-15)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.12.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.11.0...@comparethemarketau/manor-disclaimer@0.12.0) (2020-11-15)


### Features

* **slider:** fix slider end labels ([#482](https://github.com/comparethemarketau/manor-react/issues/482)) ([3ea24c2](https://github.com/comparethemarketau/manor-react/commit/3ea24c256be7fdf567186c0dfa897d0aa45a635e))





# [0.11.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.10.2...@comparethemarketau/manor-disclaimer@0.11.0) (2020-11-13)


### Features

* **disclaimer:** improvements to disclaimer style ([#480](https://github.com/comparethemarketau/manor-react/issues/480)) ([28441e5](https://github.com/comparethemarketau/manor-react/commit/28441e5f5512ff7ccadfab51b8f6ce3d77fe94da))





## [0.10.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.10.1...@comparethemarketau/manor-disclaimer@0.10.2) (2020-11-12)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.10.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.10.0...@comparethemarketau/manor-disclaimer@0.10.1) (2020-11-12)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.10.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.9.0...@comparethemarketau/manor-disclaimer@0.10.0) (2020-11-09)


### Features

* **typography:** remove nomargins pro ([#470](https://github.com/comparethemarketau/manor-react/issues/470)) ([ad14285](https://github.com/comparethemarketau/manor-react/commit/ad14285b548116be2306e19bb989113397a11e3f))





# [0.9.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.6...@comparethemarketau/manor-disclaimer@0.9.0) (2020-11-06)


### Features

* **typography:** added help text style ([#469](https://github.com/comparethemarketau/manor-react/issues/469)) ([0827882](https://github.com/comparethemarketau/manor-react/commit/0827882cd6020d5dbada7371faa19660c961af4e))





## [0.8.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.5...@comparethemarketau/manor-disclaimer@0.8.6) (2020-11-05)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.8.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.4...@comparethemarketau/manor-disclaimer@0.8.5) (2020-11-04)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.8.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.3...@comparethemarketau/manor-disclaimer@0.8.4) (2020-11-04)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.8.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.2...@comparethemarketau/manor-disclaimer@0.8.3) (2020-11-02)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.8.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.1...@comparethemarketau/manor-disclaimer@0.8.2) (2020-11-01)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.8.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.8.0...@comparethemarketau/manor-disclaimer@0.8.1) (2020-10-27)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.8.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.7.5...@comparethemarketau/manor-disclaimer@0.8.0) (2020-10-26)


### Features

* **disclaimer:** added validation message for disclaimer ([#449](https://github.com/comparethemarketau/manor-react/issues/449)) ([4b20dcc](https://github.com/comparethemarketau/manor-react/commit/4b20dcc08af7ece6920d9d20800ddf47c21afd6f))





## [0.7.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.7.4...@comparethemarketau/manor-disclaimer@0.7.5) (2020-10-26)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.7.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.7.3...@comparethemarketau/manor-disclaimer@0.7.4) (2020-10-23)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.7.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.7.2...@comparethemarketau/manor-disclaimer@0.7.3) (2020-10-22)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.7.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.7.1...@comparethemarketau/manor-disclaimer@0.7.2) (2020-10-21)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.7.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.7.0...@comparethemarketau/manor-disclaimer@0.7.1) (2020-10-20)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.7.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.12...@comparethemarketau/manor-disclaimer@0.7.0) (2020-10-19)


### Features

* **typography:** rewrote typography to no longer use materialUI ([#431](https://github.com/comparethemarketau/manor-react/issues/431)) ([6a7f7db](https://github.com/comparethemarketau/manor-react/commit/6a7f7dbef8f8bea684655f598c2ec9d55b814c34))


### BREAKING CHANGES

* **typography:** props removed: paragraph, gutterBottom. prop changes: noMargins - no defaults to
true, variant - 'button', 'inherit' options removed

* feat(typography): sronly implemented

* feat(typography): address pr feedback

* feat(typography): default display prop to 'block' instead of 'initial'

* feat(typography): update components impacted by typography changes





## [0.6.12](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.11...@comparethemarketau/manor-disclaimer@0.6.12) (2020-10-15)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.11](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.10...@comparethemarketau/manor-disclaimer@0.6.11) (2020-10-08)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.10](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.9...@comparethemarketau/manor-disclaimer@0.6.10) (2020-10-06)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.8...@comparethemarketau/manor-disclaimer@0.6.9) (2020-10-02)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.7...@comparethemarketau/manor-disclaimer@0.6.8) (2020-09-29)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.6...@comparethemarketau/manor-disclaimer@0.6.7) (2020-09-29)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.5...@comparethemarketau/manor-disclaimer@0.6.6) (2020-09-29)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.4...@comparethemarketau/manor-disclaimer@0.6.5) (2020-09-28)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.3...@comparethemarketau/manor-disclaimer@0.6.4) (2020-09-27)


### Bug Fixes

* **disclaimer:** fix disclaimer styles ([#408](https://github.com/comparethemarketau/manor-react/issues/408)) ([df31c8a](https://github.com/comparethemarketau/manor-react/commit/df31c8a623ddbdcbde533a34f94c5d8fca55f2f9))





## [0.6.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.2...@comparethemarketau/manor-disclaimer@0.6.3) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.2...@comparethemarketau/manor-disclaimer@0.6.3) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.1...@comparethemarketau/manor-disclaimer@0.6.2) (2020-09-25)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.6.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.6.0...@comparethemarketau/manor-disclaimer@0.6.1) (2020-09-24)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.6.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.14...@comparethemarketau/manor-disclaimer@0.6.0) (2020-09-24)


### Features

* **rollup:** update rollup to not bundle manor-provider ([#403](https://github.com/comparethemarketau/manor-react/issues/403)) ([0e012d6](https://github.com/comparethemarketau/manor-react/commit/0e012d6fbadcf0ec99857c22e148cacd6265b60a))





## [0.5.14](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.13...@comparethemarketau/manor-disclaimer@0.5.14) (2020-09-24)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.13](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.12...@comparethemarketau/manor-disclaimer@0.5.13) (2020-09-23)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.12](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.11...@comparethemarketau/manor-disclaimer@0.5.12) (2020-09-22)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.11](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.10...@comparethemarketau/manor-disclaimer@0.5.11) (2020-09-21)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.10](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.9...@comparethemarketau/manor-disclaimer@0.5.10) (2020-09-18)


### Bug Fixes

* **dependencies:** fix more incorrect dependencies ([#392](https://github.com/comparethemarketau/manor-react/issues/392)) ([6fa55a1](https://github.com/comparethemarketau/manor-react/commit/6fa55a11ba89125ccfe61385d9776e4185bff6f3))





## [0.5.9](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.8...@comparethemarketau/manor-disclaimer@0.5.9) (2020-09-18)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.7...@comparethemarketau/manor-disclaimer@0.5.8) (2020-09-17)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.6...@comparethemarketau/manor-disclaimer@0.5.7) (2020-09-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.5...@comparethemarketau/manor-disclaimer@0.5.6) (2020-09-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.4...@comparethemarketau/manor-disclaimer@0.5.5) (2020-09-09)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.3...@comparethemarketau/manor-disclaimer@0.5.4) (2020-09-08)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.2...@comparethemarketau/manor-disclaimer@0.5.3) (2020-09-04)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.1...@comparethemarketau/manor-disclaimer@0.5.2) (2020-09-03)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.5.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.5.0...@comparethemarketau/manor-disclaimer@0.5.1) (2020-09-01)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.5.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.4.1...@comparethemarketau/manor-disclaimer@0.5.0) (2020-08-18)


### Features

* **all:** remove manorprovider from components, fix typography props ([#365](https://github.com/comparethemarketau/manor-react/issues/365)) ([9297cf7](https://github.com/comparethemarketau/manor-react/commit/9297cf72e8a7fe8762ec0dadf07d026aa88cbb44))





## [0.4.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.4.0...@comparethemarketau/manor-disclaimer@0.4.1) (2020-08-18)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.4.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.6...@comparethemarketau/manor-disclaimer@0.4.0) (2020-08-17)


* Fix/global styles micro ui (#359) ([bcf8ce9](https://github.com/comparethemarketau/manor-react/commit/bcf8ce92ba170a51113a4022728da22f47a6a768)), closes [#359](https://github.com/comparethemarketau/manor-react/issues/359)


### BREAKING CHANGES

* Global styles have been removed from ManorProvider, as such typography is not
styled by default. Import the Font component once in your application (or macro-ui) and use the
Typography component to render text, and the Link component to render links.

* fix(fonts): fix fonts dependencies

* fix(deps): fix provider dependencies

* fix(feedback): fixes based on PR feedback

Co-authored-by: PSJgit <psjj.dev@gmail.com>





## [0.3.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.5...@comparethemarketau/manor-disclaimer@0.3.6) (2020-08-11)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.3.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.4...@comparethemarketau/manor-disclaimer@0.3.5) (2020-08-11)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.3.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.3...@comparethemarketau/manor-disclaimer@0.3.4) (2020-08-10)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.3.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.2...@comparethemarketau/manor-disclaimer@0.3.3) (2020-08-07)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.3.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.1...@comparethemarketau/manor-disclaimer@0.3.2) (2020-07-24)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.3.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.3.0...@comparethemarketau/manor-disclaimer@0.3.1) (2020-07-21)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.3.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.8...@comparethemarketau/manor-disclaimer@0.3.0) (2020-07-20)


### Features

* **providers:** remove LayerEvent/Toast providers from manorprovider ([#348](https://github.com/comparethemarketau/manor-react/issues/348)) ([461d724](https://github.com/comparethemarketau/manor-react/commit/461d72498fca1aca9de0056a27d1a3d17a89ea77))


### BREAKING CHANGES

* **providers:** drawers and modals are no longer managed by manor, users will need to use the
zIndex prop (added to both) to do so particularly when multiple layered components are rendered
simultaneously. Additionally, you will need to implement your own keyboard bindings and focus
trapping to ensure conformance to the design system.





## [0.2.8](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.7...@comparethemarketau/manor-disclaimer@0.2.8) (2020-07-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.2.7](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.6...@comparethemarketau/manor-disclaimer@0.2.7) (2020-07-14)


### Bug Fixes

* **toastmanager:** removed portals from ToastManager ([#345](https://github.com/comparethemarketau/manor-react/issues/345)) ([1533377](https://github.com/comparethemarketau/manor-react/commit/1533377910e9cbac266abe24fae1ee42eba4c52f))





## [0.2.6](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.5...@comparethemarketau/manor-disclaimer@0.2.6) (2020-06-21)


### Bug Fixes

* **imports:** fix relative imports that shouldve been package ([#338](https://github.com/comparethemarketau/manor-react/issues/338)) ([d41af0a](https://github.com/comparethemarketau/manor-react/commit/d41af0aea1f21e2fd30d281301648d44fc566da4))





## [0.2.5](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.4...@comparethemarketau/manor-disclaimer@0.2.5) (2020-06-17)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.2.4](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.3...@comparethemarketau/manor-disclaimer@0.2.4) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.2.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.2...@comparethemarketau/manor-disclaimer@0.2.3) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.2.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.1...@comparethemarketau/manor-disclaimer@0.2.2) (2020-06-15)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





## [0.2.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.2.0...@comparethemarketau/manor-disclaimer@0.2.1) (2020-06-15)

**Note:** Version bump only for package @comparethemarketau/manor-disclaimer





# [0.2.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-disclaimer@0.1.0...@comparethemarketau/manor-disclaimer@0.2.0) (2020-06-12)


### Features

* **toasts:** manorProvider now injects ToastManager ([#333](https://github.com/comparethemarketau/manor-react/issues/333)) ([8621888](https://github.com/comparethemarketau/manor-react/commit/862188867bbc8258b29fa162f46e5ad5b108f778))


### BREAKING CHANGES

* **toasts:** Added ToastNotification component. Refactored toasts such that ManorProvider
injects the ToastManager. addToasts also now expects a <ToastNotification> instead of an object.

* fix(story): fix typo in toast notification story





# 0.1.0 (2020-06-11)


### Features

* **packages:** migrated to monorepo ([920a45e](https://github.com/comparethemarketau/manor-react/commit/920a45ec4b40a19de32f39f29693cbe1b1f314ae))
