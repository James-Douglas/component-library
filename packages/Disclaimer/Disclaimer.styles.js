import styled from 'styled-components';

export const StyledWrapper = styled.div`
  display: flex;
  align-items: flex-start;
`;

export const StyledContent = styled.div`
  margin-top: 0.6rem;
  margin-left: ${({ theme }) => theme.spacing[12]};
  cursor: pointer;
  font-size: ${({ theme }) => theme.fontSize.base};
`;
