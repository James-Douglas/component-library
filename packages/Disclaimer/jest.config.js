const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Disclaimer',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Disclaimer',
  displayName: 'Disclaimer',
  rootDir: '../..',
};
