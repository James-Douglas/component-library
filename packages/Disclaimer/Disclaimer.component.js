import React, {
  createRef, useEffect, useState,
} from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@comparethemarketau/manor-typography';
import { useId } from '@comparethemarketau/manor-hooks';
import { Checkbox } from '@comparethemarketau/manor-checkbox';
import { StyledContent, StyledWrapper } from './Disclaimer.styles';

const Disclaimer = ({
  id: propsId, isSelected, children, handleChange, handleFocus, handleBlur,
}) => {
  const id = useId(propsId);
  const content = createRef();
  const [checked, setChecked] = useState(isSelected);

  useEffect(() => {
    setChecked(isSelected);
  }, [isSelected]);

  const changeHandler = () => {
    setChecked(!checked);
    if (handleChange) {
      handleChange(!checked);
    }
  };

  const handleContentClick = (e) => {
    if (e.target.nodeName.toLowerCase() !== 'a') {
      if (e.target === content.current || content.current.contains(e.target)) {
        const newChecked = !checked;
        setChecked(newChecked);
        if (handleChange) {
          handleChange(newChecked);
        }
      }
    }
  };

  return (
    <StyledWrapper>
      <Checkbox id={id} handleChange={changeHandler} isSelected={checked} handleFocus={handleFocus} handleBlur={handleBlur} />
      <StyledContent ref={content} onClick={handleContentClick}>
        <Typography variant="body1">
          {children}
        </Typography>
      </StyledContent>
    </StyledWrapper>
  );
};

Disclaimer.propTypes = {
  /**
   * Unique identifier for the disclaimers' checkbox
   */
  id: PropTypes.string,
  /**
   * Defines the Disclaimers checked state.
   */
  isSelected: PropTypes.bool,
  /**
   * The content to be displayed next to the Checkbox
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  /**
   * Called on change with the value of the Disclaimers checkbox
   */
  handleChange: PropTypes.func,
  /**
   * Called on focus of the checkbox
   */
  handleFocus: PropTypes.func,
  /**
   * Called on blur of the checkbox
   */
  handleBlur: PropTypes.func,
};

Disclaimer.defaultProps = {
  id: null,
  children: [],
  isSelected: false,
  handleChange: null,
  handleFocus: null,
  handleBlur: null,
};

export default Disclaimer;
