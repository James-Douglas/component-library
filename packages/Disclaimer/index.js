import Disclaimer from './Disclaimer.component';

export { Disclaimer };
