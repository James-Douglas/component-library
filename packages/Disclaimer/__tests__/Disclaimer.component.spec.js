import React from 'react';
import { render, fireEvent } from '../../../testUtils';
import Disclaimer from '../Disclaimer.component';

describe('Disclaimer', () => {
  it('renders with minimal props', () => {
    const { container } = render(<Disclaimer trackingLabel="test" id="test" />);
    expect(container.innerHTML).toMatchSnapshot();
  });
  it('checks on click', () => {
    const changeHandler = jest.fn();
    const { container } = render(<Disclaimer trackingLabel="test" id="test" handleChange={changeHandler} />);
    const checkbox = container.querySelector('#test');
    fireEvent.click(checkbox);
    expect(checkbox.checked).toBe(true);
    expect(changeHandler).toHaveBeenCalled();
    expect(changeHandler.mock.calls[0][0]).toBe('test', true);
  });
  it('checks on click of label', () => {
    const changeHandler = jest.fn();
    const { container, getByText } = render(<Disclaimer trackingLabel="test" id="test-2" handleChange={changeHandler}>test disclaimer text</Disclaimer>);
    const checkbox = container.querySelector('#test-2');
    const disclaimerText = getByText('test disclaimer text');
    fireEvent.click(disclaimerText);
    expect(checkbox.checked).toBe(true);
    expect(changeHandler).toHaveBeenCalled();
    expect(changeHandler.mock.calls[0][0]).toBe('test-2', true);
  });
});
