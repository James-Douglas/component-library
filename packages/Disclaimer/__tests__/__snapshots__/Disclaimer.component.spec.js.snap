// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer renders with minimal props 1`] = `"<div class=\\"sc-AxjAm kaSHsU\\"><div class=\\"sc-AxirZ jnDRS\\"></div></div><div class=\\"sc-fzpmMD kGLgCC\\"><div class=\\"sc-AxheI fPdsej\\"><input id=\\"test\\" name=\\"test\\" type=\\"checkbox\\" role=\\"checkbox\\" aria-checked=\\"false\\" class=\\"sc-AxgMl kGkIyX\\"><label for=\\"test\\" class=\\"sc-AxmLO bPwmMK\\"><div class=\\"sc-Axmtr hPdpgm\\"></div></label></div><div class=\\"sc-fznxKY jvCjjU\\"></div></div>"`;
